packages/message-manager/src/MessageManager.ts

```diff
@@ -66,16 +70,13 @@ export class MessageManager extends AbstractMessageManager<
    * Creates a new Message with an 'unapproved' status using the passed messageParams.
    * this.addMessage is called to add the new Message to this.messages, and to save the unapproved Messages.
    *
-   * @param messageParams - The params for the eth_sign call to be made after the message is approved.
-   * @param req - The original request object possibly containing the origin.
+   * @param messageParamsWithId - The params for the eth_sign call to be made after the message is approved.
    * @returns Promise resolving to the raw data of the signature request.
    */
-  async addUnapprovedMessageAsync(
-    messageParams: MessageParams,
-    req?: OriginalRequest,
+  async createMessageListener(
+    messageParamsWithId: MessageParamsWithId,
   ): Promise<string> {
-    validateSignMessageData(messageParams);
```

Comment:
As we're removing the validation, this method is almost identical in all three message managers.

I appreciate the strings include the message name, plus the `TypedMessageManager` has the error status, but is it worth defining this once in the `AbstractMessageManager`? Something like:

 `waitForFinishStatus(messageManager, messageName, hasErrorState)`?

Reply:
Fixed, moved into AMM